pkg/distributors/contracts/RewardsScheduler.sol

```diff
@@ -0,0 +1,118 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+import "@balancer-labs/v2-solidity-utils/contracts/openzeppelin/SafeERC20.sol";
+import "./interfaces/IMultiRewards.sol";
+
+// solhint-disable not-rely-on-time
+
+/**
+ * Scheduler for MultiRewards contract
+ */
+contract RewardsScheduler {
+    using SafeERC20 for IERC20;
+
+    IMultiRewards public multirewards;
+
+    constructor(IMultiRewards _multirewards) {
+        multirewards = _multirewards;
+    }
+
+    struct ScheduledReward {
+        IERC20 pool;
+        IERC20 rewardsToken;
+        uint256 startTime;
+        address rewarder;
+        uint256 amount;
+    }
+
+    event RewardScheduled(bytes32 rewardId, address scheduler, address rewardsToken, uint256 startTime);
+    event RewardUnscheduled(bytes32 rewardId, address scheduler, address rewardsToken, uint256 startTime);
+
+    mapping(bytes32 => ScheduledReward) private _rewards;
+
+    function poke(bytes32[] calldata rewardIds) external {
+        for (uint256 r; r < rewardIds.length; r++) {
+            bytes32 rewardId = rewardIds[r];
+            ScheduledReward memory scheduledReward = _rewards[rewardId];
+
+            require(scheduledReward.startTime != 0, "reward has not been created");
+            require(scheduledReward.startTime <= block.timestamp, "reward cannot be started");
+
+            multirewards.startScheduledReward(
+                scheduledReward.pool,
+                scheduledReward.rewardsToken,
+                scheduledReward.amount,
+                scheduledReward.rewarder
+            );
+            delete _rewards[rewardId];
```

Comment:
I'd not delete this entry, and instead a) mark it as completed (maybe have an enum with `uninit = 0, pending, completed` instead of using start time == 0), and b) add a getter `getRewardInfo(rewardId)` so that past rewards can be more easily queried.

Reply:
What's the use case of querying past reward schedules on-chain?  I didn't opt to save them for simplicity/cleanup

By the same logic should we be saving past records of rewards added to the staking contract itself?  Those seem like arguably the more important ones to save.

Reply:
Well, we didn't introdce many changes in MultiRewards precisely to avoid changes, so I wouldn't use it as a basis for comparison.

My point about data availability was more that it is strange for some piece of data to be available and then deleted after it is 'used' - I've seen projects be bitten by this, and in general try to avoid `delete` altogether.

Reply:
I disagree, but I've made the change